lib/mongoid/contextual/map_reduce.rb

```diff
@@ -147,7 +147,7 @@ def output
       def raw
         validate_out!
         cmd = command
-        opts = { read: cmd.delete(:read) } if cmd[:read]
+        opts = { read: criteria.options.fetch(:read) } if criteria.options[:read]
```

Comment:
Thank you for this, I gather this repairs failures that I've seen in another PR.